### PR TITLE
[FX-1395] Formally delete `StopgapGlobalState`

### PR DESCRIPTION
**Pull request by devinmorgan**
## What
Start passing down the `envConfig` value as a param to PinSubmitter-related facilities and stop reading the `envConfig` value from global mutable state.

## Why
`StopgapGlobalState` was a temporary workaround to promptly ship v3 of this SDK. We want to remove it to:
1. reduce the possibility of global state-related bugs and
2. to give the Pos Sdk UI hierarchy additional flexibility to support the `ForagePinPad`.

## Test Plan
- ❌ Did not add tests since these changes are on the view-layer
- ❌ No need to manually test; passing CI tests are sufficient

## Demo
N/A since this only changes implementation details

## How
Can be released once #252 is released.

### CODE REVIEW

**Comment by devinmorgan** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/253?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#253** <a href="https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/253?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#252** <a href="https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/252?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @devinmorgan and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by devinmorgan** [comment]
### Merge activity

* **Jun 13, 4:51 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/253) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

File: forage-android/src/test/java/com/joinforage/forage/android/vault/BasisTheoryPinSubmitterTest.kt
Comment (djoksimo) [inline forage-android/src/test/java/com/joinforage/forage/android/vault/BasisTheoryPinSubmitterTest.kt:1]: Why was this file removed 😢

**Review by djoksimo — CHANGES_REQUESTED** [review]
Thank you for making this long-awaited change @devinmorgan 🤩

Left a question about why Basis Theory tests were removed, but otherwise LGTM!

File: forage-android/src/test/java/com/joinforage/forage/android/vault/BasisTheoryPinSubmitterTest.kt
Comment (devinmorgan) [reply forage-android/src/test/java/com/joinforage/forage/android/vault/BasisTheoryPinSubmitterTest.kt:1]: When I deleted `StopgapGlobalState,` this test suite broke. I spent about an hour trying to get it to work to no avail and wanted to prioritize making sure we hit our 6/20 release. But you are right to point out the importance of test coverage. Would it be alright for now that I create [this ticket](https://linear.app/joinforage/issue/FX-1512/bring-back-test-coverage-to-pin-submitter-facillities) to make sure that I bring back coverage to these facilities for after the release?

**Review by djoksimo — APPROVED** [review]
I will add a follow-up PR on top of your stack to restore Basis Theory PIN Submitter tests
